subsys/net/lib/utils/addr_utils.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2019 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <net/socket.h>
+
+/* These utility functions are intended to be on top of POSIX API, and don't
+ * make sense if it's not available.
+ */
+#ifdef CONFIG_NET_SOCKETS_POSIX_NAMES
+
+const char *net_addr_str_find_port(const char *addr_str)
+{
+	const char *p = strrchr(addr_str, ':');
+
+	if (p == NULL) {
+		return NULL;
+	}
+
+	/* If it's not IPv6 numeric notation, we guaranteedly got a port */
+	if (*addr_str != '[') {
+		return p + 1;
+	}
+
+	/* IPv6 numeric address, and ':' preceeded by ']' */
+	if (p[-1] == ']') {
+		return p + 1;
+	}
+
+	/* Otherwise, just raw IPv6 address, ':' is component separator */
+	return NULL;
+}
+
+int net_getaddrinfo_addr_str(const char *addr_str, const char *def_port,
```

Comment:
> As the functionality is almost identical to getaddrinfo(), why not call it net_getaddrinfo_str() as the addr looks redundant information.

The idea was to provide enough information and differentiation from raw getaddrinfo(), while keep it reasonably short. In that regard, as string in format `host[:port]` is called "address string", abbreviated `addr_str`, and that infix is used consistently in both function names, net_addr_str_find_port() and net_getaddrinfo_addr_str().

Even more selective name, like `addr_str_w_port`, would be better, but would lead to long function name. OTOH, just using "str" isn't selective enough IMHO.